core/src/Spirit/Chain.cpp

```diff
@@ -9,710 +9,726 @@
-    std::shared_ptr<Data::Spin_System_Chain> chain;
-    
-    // Fetch correct indices and pointers
     try
     {
-        from_indices( state, idx_image, idx_chain, image, chain );
+        if (idx_chain>=0) return state->collection->chains[idx_chain]->noi;
+        return state->active_chain->noi;
     }
-    catch( const Utility::Exception & ex )
+    catch( ... )
     {
-        Utility::Handle_exception( ex, idx_image, idx_chain );
-        return false;
+        Utility::Handle_Exception( -1, idx_chain );
+        return 0;
     }
-    
-    // Apply
-    ++chain->idx_active_image;
-    State_Update(state);
-    
-    Log( Utility::Log_Level::Debug, Utility::Log_Sender::API,
-         "Switched to next image " + std::to_string(chain->idx_active_image+1) + " of " + 
-         std::to_string(chain->noi), chain->idx_active_image, idx_chain );
-    
-    return true;
 }
 
-bool Chain_prev_Image(State * state, int idx_chain_i)
+bool Chain_next_Image(State * state, int idx_chain)
 {
-    int idx_image = -1, idx_chain = idx_chain_i;
-    std::shared_ptr<Data::Spin_System> image;
-    std::shared_ptr<Data::Spin_System_Chain> chain;
+    int idx_image = -1;
     
-    // Fetch correct indices and pointers
     try
     {
+        std::shared_ptr<Data::Spin_System> image;
+        std::shared_ptr<Data::Spin_System_Chain> chain;
+        
+        // Fetch correct indices and pointers
         from_indices( state, idx_image, idx_chain, image, chain );
-    }
-    catch( const Utility::Exception & ex )
-    {
-        Utility::Handle_exception( ex, idx_image, idx_chain );
-        return false;
-    }
-    
-    // Apply
-    if ( idx_image > 0 )
-    {
-        --chain->idx_active_image;
+        
+        // Apply
+        ++chain->idx_active_image;
         State_Update(state);
-        Log(Utility::Log_Level::Debug, Utility::Log_Sender::API,
-            "Switched to previous image " + std::to_string(chain->idx_active_image+1) + " of " + std::to_string(chain->noi), chain->idx_active_image, idx_chain);
+        
+        Log( Utility::Log_Level::Debug, Utility::Log_Sender::API,
+             "Switched to next image " + std::to_string(chain->idx_active_image+1) + " of " + 
+             std::to_string(chain->noi), chain->idx_active_image, idx_chain );
+        
         return true;
     }
-    else
+    catch( ... )
     {
-        Log(Utility::Log_Level::Error, Utility::Log_Sender::API,
-            "Tried to switch to previous image.", chain->idx_active_image, idx_chain);
+        Utility::Handle_Exception( idx_image, idx_chain );
         return false;
     }
 }
 
-bool Chain_Jump_To_Image(State * state, int idx_image_i, int idx_chain_i)
+bool Chain_prev_Image( State * state, int idx_chain )
 {
-    int idx_image = idx_image_i, idx_chain = idx_chain_i;
-    std::shared_ptr<Data::Spin_System> image;
-    std::shared_ptr<Data::Spin_System_Chain> chain;
-    
-    // Fetch correct indices and pointers
+    int idx_image = -1;
```

Comment:
It is probably not a problem to have the definition of `idx_image` outside of the `try{ ... } catch{ ... }` (so to have it available in both scopes) but a better way would be to change argument's order in `Utility::Handle_Exception()` function [(see that comment)](https://github.com/spirit-code/spirit/pull/281/commits/c173dd19f953859a952baf8c82e0c0ca089bb520#r132786843)